core/src/main/java/com/scalar/db/util/groupcommit/NormalGroup.java

```diff
@@ -0,0 +1,181 @@
+      KeyManipulator<PARENT_KEY, CHILD_KEY, FULL_KEY, EMIT_KEY> keyManipulator) {
+    super(emitter, keyManipulator, config.slotCapacity(), config.oldGroupAbortTimeoutSeconds());
+    this.delayedSlotMoveTimeoutMillis = config.delayedSlotMoveTimeoutMillis();
+    this.groupSizeFixTimeoutAtMillis =
+        System.currentTimeMillis() + config.groupSizeFixTimeoutMillis();
+    updateDelayedSlotMoveTimeoutAt();
+    this.parentKey = keyManipulator.generateParentKey();
+  }
+
+  PARENT_KEY parentKey() {
+    return parentKey;
+  }
+
+  @Override
+  FULL_KEY fullKey(CHILD_KEY childKey) {
+    return keyManipulator.fullKey(parentKey, childKey);
+  }
+
+  // If it returns null, the Group is already size-fixed and a retry is needed.
+  @Nullable
+  FULL_KEY reserveNewSlot(CHILD_KEY childKey) {
+    return reserveNewSlot(new Slot<>(childKey, this));
+  }
+
+  @Nullable
+  synchronized List<Slot<PARENT_KEY, CHILD_KEY, FULL_KEY, EMIT_KEY, V>> removeNotReadySlots() {
+    if (!isSizeFixed()) {
+      logger.info(
+          "No need to remove any slot since the size isn't fixed yet. Too early. Group: {}", this);
+      return null;
+    }
+
+    // Lazy instantiation might be better, but it's likely there is a not-ready value slot since
+    // it's already timed-out.
+    List<Slot<PARENT_KEY, CHILD_KEY, FULL_KEY, EMIT_KEY, V>> removed = new ArrayList<>();
+    for (Entry<CHILD_KEY, Slot<PARENT_KEY, CHILD_KEY, FULL_KEY, EMIT_KEY, V>> entry :
+        slots.entrySet()) {
+      Slot<PARENT_KEY, CHILD_KEY, FULL_KEY, EMIT_KEY, V> slot = entry.getValue();
+      if (!slot.isReady()) {
+        removed.add(slot);
+      }
+    }
+
+    // The size must be already fixed since the group is already size-fixed.
+    Integer size = size();
+    assert size != null;
+    if (removed.size() >= size) {
+      logger.debug("No need to remove any slot since all the slots are not ready. Group: {}", this);
+      return null;
+    }
+
+    for (Slot<PARENT_KEY, CHILD_KEY, FULL_KEY, EMIT_KEY, V> slot : removed) {
+      removeSlot(slot.key());
+      logger.debug(
+          "Removed a value slot from group to move it to delayed group. Group: {}, Slot: {}",
+          this,
+          slot);
+    }
+    return removed;
+  }
+
+  @Override
+  public synchronized void delegateEmitTaskToWaiter() {
+    final AtomicReference<Slot<PARENT_KEY, CHILD_KEY, FULL_KEY, EMIT_KEY, V>> emitterSlot =
+        new AtomicReference<>();
+
+    boolean isFirst = true;
+    List<V> values = new ArrayList<>(slots.size());
+    for (Slot<PARENT_KEY, CHILD_KEY, FULL_KEY, EMIT_KEY, V> slot : slots.values()) {
+      // Use the first slot as an emitter.
+      if (isFirst) {
+        isFirst = false;
+        emitterSlot.set(slot);
+      }
+      values.add(slot.value());
+    }
+
+    // This task is passed to only the first slot, so the slot will be resumed.
+    // Other slots will be blocked until `markAsXxxx()` is called.
+    ThrowableRunnable<GroupCommitException> taskForEmitterSlot =
```

Comment:
```suggestion
    ThrowableRunnable<Exception> taskForEmitterSlot =
```

Reply:
Thanks! Fixed it in https://github.com/scalar-labs/scalardb/pull/1607/commits/865830013b893112149fa88ce705cd60aa465226